test/e2e/es_cluster_logging.go

```diff
@@ -0,0 +1,293 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Cluster level logging using Elasticsearch", func() {
+	var c *client.Client
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should check that logs from pods on all nodes are ingested into Elasticsearch", func() {
+		ClusterLevelLoggingWithElasticsearch(c)
+	})
+})
+
+func bodyToJSON(body []byte) (map[string]interface{}, error) {
+	var r interface{}
+	if err := json.Unmarshal(body, &r); err != nil {
+		Logf("Bad JSON: %s", string(body))
+		return nil, fmt.Errorf("failed to unmarshal Elasticsearch response: %v", err)
+	}
+	rJSON, ok := r.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("failed to convert JSON response to map[string]interface{}: %v", r)
+	}
+	return rJSON, nil
+}
+
+// ClusterLevelLoggingWithElasticsearch is an end to end test for cluster level logging.
+func ClusterLevelLoggingWithElasticsearch(c *client.Client) {
+
+	// For now assume we are only testing cluster logging with Elasticsearch
+	// on GCE. Once we are sure that Elasticsearch cluster level logging
+	// works for other providers we should widen this scope of this test.
+	if testContext.provider != "gce" {
+		Logf("Skipping cluster level logging test for provider %s", testContext.provider)
+		return
+	}
+
+	// Check to see if we can speak to the Elasticsearch service.
+	By("Connecting to Elasticsearch service.")
+	ns := api.NamespaceDefault
+	s := c.Services(ns)
+	_, err := s.Get("elasticsearch-logging")
```

Comment:
I wondered about that but I think service creation (if there is no load balancer) is quick so I thought we would try it without such a loop and only add a wait for service available check if we need it.